### ISSUE TITLE
Search docs should mention that you need BasePageController

### ISSUE THREAD

**Issue by chillu** [silverstripe/cwp#202]
Looking at the search docs, there's an expectation that you're using `PageController extends BasePageController`, which is the default through `cwp/installer`. `BasePage` pulls in a lot of stuff that can be unnecessary for CWP projects (e.g. "related pages"), and many CWP devs will remove it from their own inheritance structures.

Which is an issue if you're then following the search docs, because the CWP default search behaviour is configured through:

```yml
---
Name: cwpsearchextensions
---
CWP\CWP\PageTypes\BasePageController:
  extensions:
    - CWP\Search\Extensions\SearchControllerExtension
```

Unless you have this installed, you won't see a search box in your Watea theme. This chain of events is so hard to follow that devs will likely spend hours debugging why their search isn't showing. We should mention that dependency in the CWP search docs.

**Comment by robbieaverill** [comment]
>This chain of events is so hard to follow that devs will likely spend hours debugging why their search isn't showing.

I've done that multiple times
